ambari-server/src/main/java/org/apache/ambari/server/security/encryption/EncryptionUtils.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ambari.server.security.encryption;
+
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+
+import org.apache.commons.codec.binary.Base64;
+
+import com.google.inject.Singleton;
+
+@Singleton
+public class EncryptionUtils {
+
+  private static final String BASE_64_FIELD_DELIMITER = "::";
+  private static final String UTF_8_CHARSET = StandardCharsets.UTF_8.name();
+
+  private MasterKeyService environmentMasterKeyService;
+
+  /**
+   * Encrypts the given text using the master key found in the environment
+   * 
+   * @param toBeEncrypted
+   *          the text to be encrypted
+   * @return the String representation of the encrypted text
+   * @throws Exception
+   *           in case any error happened during the encryption process
+   */
+  public String encrypt(String toBeEncrypted) throws Exception {
+    return encrypt(toBeEncrypted, (String) null);
+  }
+
+  /**
+   * Encrypts the given text using the given master key
+   * 
+   * @param toBeEncrypted
+   *          the text to be encrypted
+   * @param masterKey
+   *          the mater key to be used for encryption
+   * @return the String representation of the encrypted text
+   * @throws Exception
+   *           in case any error happened during the encryption process
+   */
+  public String encrypt(String toBeEncrypted, String masterKey) throws Exception {
```

Comment:
The caller should pass in the _key_ to use... not the "master key" which is the key known to the Ambari server.  Maybe rename this to "key".  It is possible that some other key will be used to encrypt the `toBeEncrypted` string.  We should not tie this utility class directly to Ambari's master key.

Reply:
ok